### DISCUSSION TITLE
[Bug]: Reconcilation process for strimzi cluster operator multiple times happening

### DISCUSSION THREAD

**Discussion by nsettipa** [strimzi/strimzi-kafka-operator#10939] [General]
### Bug Description

When new environment variables or labels added to Kafka or cruisecontrol or mirror maker. The newly added environment getting reconciled in the 2nd attempt causing restart of Kafka pods more than once, this is causing upgrade time from one version to another version going more than 13 or 15 minutes as all the Pods have to brought up 2nd time to have new configuration in place. 

Is there possibility for upgrade scenarios to reconcile all the newly added configurations such as environment variables or custom data in the 1st reconcilation itself. 

### Expected behavior

Would like to see improvement for upgrade scenarios. Please provide anything we can do in the application to consider all the configuration in-place before 1st reconcilation.

### Strimzi version

1.27

### Kubernetes version

Kubernetes 

### Installation method

upgrade from one version to another version where kafka broker pods, mirror maker and cruise-control exists along with entity operator

### Infrastructure

Baremetal configuration

### Additional context

Provided all the information

**Comment by scholzj** [comment]
I think you should provide some logs, examples and reproducers steps to help understand what exactly are you talking about. Because from the description it is not really clear.
